### PR TITLE
cleaned and fixed mash.rule

### PR DESCRIPTION
**Pull request by RNAer**
also cleaned up the master Snakefile. from #57 

Just to be clear, PR against dev doesn't need to wait for travis to merge. waiting to be further reviewed and merged. @wasade @tanaes

### CODE REVIEW

File: oecophylla/distance/mash.rule [hunk 1]
```diff
@@ -1,99 +1,51 @@
...
+        forward = qc_dir + "{sample}/filtered/{sample}.R1.trimmed.filtered.fastq.gz"
     output:
-        sketch = distance_dir + '{sample}/mash/{sample}.msh'
+        distance_dir + '{sample}/mash/{sample}.msh'
     params:
-        mash = config['software']['mash'],
-        seqtk = config['software']['seqtk'],
-        mash_params = config['params']['mash']['other'],
-        depth = config['params']['mash']['depth']
+        env = config['envs']['distance'],
```
Comment (wasade) [inline oecophylla/distance/mash.rule:10]: trailing comma?

File: oecophylla/distance/mash.rule [hunk 2]
```diff
@@ -1,99 +1,51 @@
...
-        for i in range(len(input)):
-            for j in range(i,len(input)):
-                thing1 = input[i]
-                thing2 = input[j]
-                shell("""
-                      {params.mash} dist {thing1} {thing2} >> {output.dm}
-                      """)
+        from itertools import combinations
+        for i, j in combinations(input, 2):
+            shell('mash dist {i} {j} >> {output.dm}')
```
Comment (wasade) [inline oecophylla/distance/mash.rule:42]: is this expensive? might be a fun target for GNU `parallel`

File: oecophylla/distance/mash.rule (hunk 1 above)
Comment (RNAer) [reply oecophylla/distance/mash.rule:10]: no. it is required between lines or variables

File: oecophylla/distance/mash.rule (hunk 2 above)
Comment (RNAer) [reply oecophylla/distance/mash.rule:42]: should not be expensive. basically comparing the hashes of a small number. KISS for now

**Review by tanaes — APPROVED** [review]
This is nicely cleaned up. Thanks!

**Pull request merged**